includes/REST/OrderControllerV2.php

```
@@ -138,38 +145,27 @@ public function get_order_downloads( $request ) {
-                    );
-                    $download->product = reset( $filter_items );
-
-                    return $this->prepare_data_for_response( $download, $request );
-            },
-            $downloads
-        );
+        // Filter downloads with existing products and prepare response.
+        $downloads = [];
+        foreach ( $download_permissions as $download ) {
+            $product_id = intval( $download->product_id );
+            if ( isset( $products_by_id[ $product_id ] ) ) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$products_by_id'. (undefined)

(UnusedLocalVariable)